emulator.c

```diff
@@ -39,6 +39,12 @@ execute_instruction(i8080 *cpu, uint8_t opcode)
       }
     case 0x09: // NOLINT
       printf("DAD B");
+      uint32_t bc = (cpu->b << 8) | (cpu->c); // NOLINT
```

Comment:
I think bc and hl can be unit32_t. Result should be unit32_t so that the > 0xFF check works.